### PR TITLE
gwt-helloworld

### PR DESCRIPTION
**Pull request by csadilek**
This pull request contains the gwt-helloworld quickstart app. It demonstrates the use of CDI and JAX-RS in JBoss AS 7 with a GWT front-end client.

### CODE REVIEW

File: gwt-helloworld/pom.xml [hunk 1]
```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    
+    <groupId>org.jboss.as.quickstarts</groupId>
+    <artifactId>gwt-helloworld</artifactId>
```
Comment (pmuir) [inline gwt-helloworld/pom.xml]: Normally we use jboss-as-<name> here.

File: gwt-helloworld/pom.xml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        <artifactId>gwt-user</artifactId>
+        <version>2.4.0</version>
+        <scope>provided</scope>
+      </dependency>
+    </dependencies>
+    
+    <build>
+      <!-- Set the name of the war, used as the context root when the app is 
+        deployed -->
+      <finalName>gwt-helloworld</finalName>
```
Comment (pmuir) [inline gwt-helloworld/pom.xml]: Normally we use ${project.artifactId} here, to ensure that WTP doesn't have problems.

File: gwt-helloworld/pom.xml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            <webappDirectory>war</webappDirectory>
+            <packagingExcludes>**/client/local/**/*.class</packagingExcludes>
+          </configuration>
+        </plugin>
+    
+        <!-- JBoss AS plugin to deploy war -->
+        <plugin>
+          <groupId>org.jboss.as.plugins</groupId>
+          <artifactId>jboss-as-maven-plugin</artifactId>
+          <version>7.0.2.Final</version>
```
Comment (pmuir) [inline gwt-helloworld/pom.xml]: This needs to be 7.1.0.Beta1b

File: gwt-helloworld/pom.xml [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      <finalName>gwt-helloworld</finalName>
+      <plugins>
+        <plugin>
+          <artifactId>maven-war-plugin</artifactId>
+          <version>2.1.1</version>
+          <configuration>
+            <!-- Java EE 6 doesn't require web.xml, Maven needs to catch up! -->
+            <failOnMissingWebXml>false</failOnMissingWebXml>
+            <warSourceDirectory>war</warSourceDirectory>
+            <webappDirectory>war</webappDirectory>
```
Comment (pmuir) [inline gwt-helloworld/pom.xml]: We should have a comment about why we are overriding these items.

File: gwt-helloworld/pom.xml [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            <packagingExcludes>**/client/local/**/*.class</packagingExcludes>
```
Comment (pmuir) [inline gwt-helloworld/pom.xml]: What is this for? Should have a comment.

File: gwt-helloworld/pom.xml [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            <runTarget>HelloWorldApp.html</runTarget>
+            <warSourceDirectory>war</warSourceDirectory>
+            <extraJvmArgs>-Xmx512m</extraJvmArgs>
+            <hostedWebapp>war</hostedWebapp>
+            <noServer>true</noServer>
+            <runTarget>http://localhost:8080/gwt-helloworld</runTarget>
+          </configuration>
+          <executions>
+            <execution>
+              <id>gwt-compile</id>
```
Comment (pmuir) [inline gwt-helloworld/pom.xml]: A quick comment here would help.

File: gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/HelloWorldApp.gwt.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE module PUBLIC "-//Google Inc.//DTD Google Web Toolkit 1.6//EN"
+   "http://google-web-toolkit.googlecode.com/svn/releases/1.6/distro-source/core/src/gwt-module.dtd">
+<module rename-to="HelloWorldApp">
```
Comment (pmuir) [inline gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/HelloWorldApp.gwt.xml]: Comment.

File: gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/client/local/HelloResponse.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+ * JAX-RS HelloWorldResource service}.
+ * 
+ * @author Jonathan Fuerth <jfuerth@redhat.com>
+ * @author Christian Sadilek <csadilek@redhat.com>
+ */
+public class HelloResponse extends JavaScriptObject {
+
+  protected HelloResponse() {}
+  
+  public final native String getResult() /*-{ return this.result; }-*/;
```
Comment (pmuir) [inline gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/client/local/HelloResponse.java]: Maybe a quick explanation of what's going on here?

**Comment by pmuir** [comment]
Few minor points, but looking good!

File: gwt-helloworld/pom.xml (hunk 3 above)
Comment (jfuerth) [inline gwt-helloworld/pom.xml]: We couldn't get the jboss-as:deploy to work on that version. We'll try again and follow up if it's still not working for us.

**Comment by jfuerth** [comment]
Thanks for the review. All good points. We'll work on the fixes right away.

You, sir, are a machine. :)

File: gwt-helloworld/pom.xml (hunk 1 above)
Comment (csadilek) [inline gwt-helloworld/pom.xml]: done.

File: gwt-helloworld/pom.xml (hunk 2 above)
Comment (csadilek) [inline gwt-helloworld/pom.xml]: done.

File: gwt-helloworld/pom.xml (hunk 3 above)
Comment (csadilek) [inline gwt-helloworld/pom.xml]: It worked fine now, so we are using 7.1.0.Beta1b.

File: gwt-helloworld/pom.xml (hunk 4 above)
Comment (csadilek) [inline gwt-helloworld/pom.xml]: war is the GWT default output dir. We decided not to override them anymore but use src/main/webapp instead. This should be easier to use with JBoss tools + we don't need to explain the additional items.

File: gwt-helloworld/pom.xml (hunk 5 above)
Comment (csadilek) [inline gwt-helloworld/pom.xml]: Added a comment. Basically, we are excluding client only classes from the deployment. They get compiled down to JavaScript anyway and would only introduce additional dependencies an runtime.

File: gwt-helloworld/pom.xml (hunk 6 above)
Comment (csadilek) [inline gwt-helloworld/pom.xml]: Simplified it and added a comment.

File: gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/HelloWorldApp.gwt.xml (hunk 7 above)
Comment (csadilek) [inline gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/HelloWorldApp.gwt.xml]: Done :).

File: gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/client/local/HelloResponse.java (hunk 8 above)
Comment (csadilek) [inline gwt-helloworld/src/main/java/org/jboss/as/quickstarts/gwthelloworld/client/local/HelloResponse.java]: Done too.

**Comment by csadilek** [comment]
Thanks for your help, Pete. All changes should be in now.

**Comment by pmuir** [comment]
Sande, can you do a review?

**Pull request merged**